File: cursos/pria/src/csharp/1_Variables.cs
// Esto es un comentario de una línea

/*
   Esto es un comentario
   de múltiples líneas
*/

// VARIABLES
// Una variable es un símbolo que representa una cantidad que puede variar.
// En C#, la sintaxis general para declarar una variable es:
// tipoDeDato nombreVariable = valor;

int edad = 25;  // La edad es un número entero, por eso usamos "int"



// TIPOS DE DATOS BÁSICOS

int nivelDelMar = 25;              // Entero (Integer)
float temperatura = -3.82f;        // Número real (Float)
string nombre = "Nacho López";     // Cadena de texto (String)
bool tieneCoche = true;            // Booleano (Boolean): solo puede ser 'true' o 'false'



// OPERACIONES ARITMÉTICAS CON NÚMEROS

int x = 5;
int y = 2;

int z;
z = x + y;          // Suma.                                 Resultado: 7
z = x - y;          // Resta.                                Resultado: 3
z = x * y;          // Multiplicación.                       Resultado: 10
z = x / y;          // División.                             Resultado: 2 (Nota: la división entre enteros redondea al entero más cercano)
z = x % y;          // Módulo (resto de la división).        Resultado: 1

z = z + 1;          // Incrementa el valor de z en 1.        Resultado: 2
z = z - 1;          // Disminuye el valor de z en 1.         Resultado: 1

// Nota: En la división, si deseas obtener un resultado con decimales, uno de los números debe ser de tipo float o double:
float resultadoDivision = 5.0f / 2.0f;  // Resultado: 2.5



// OPERACIONES BÁSICAS CON CADENAS DE TEXTO (strings)

string a = "GNU/";
string b = "Linux";
string c = a + b;  // Concatenación. Resultado: "GNU/Linux"


// IMPRIMIR VARIABLES EN LA PANTALLA

Console.WriteLine("¡Hola, mundo!");   // Imprime en la pantalla: ¡Hola, mundo!
Console.WriteLine(x);                 // Imprime el valor de x

// Puedes imprimir en la pantalla cadenas de texto y variables
Console.WriteLine("He comprado " + x + " naranjas y " + y + " limones.");



// CONVERSIÓN DE TIPOS DE DATOS

string posicion = "5";
string calorias = "95.4";
int peso = 85;
float altitud = -544.432f;

int posicionInt1 = Convert.ToInt32(posicion);     // Convertir a entero
float caloriasFlt1 = Convert.ToSingle(calorias);  // Convertir a número real (float)
string pesoStr1 = Convert.ToString(peso);         // Convertir a cadena de texto (string)
string altitudStr1 = Convert.ToString(altitud);   // Convertir a cadena de texto (string)

int posicionInt2 = int.Parse(posicion);      // Convertir a entero
float caloriasFlt2 = float.Parse(calorias);  // Convertir a número real (float)
string pesoStr2 = peso.ToString();           // Convertir a cadena de texto (string)
string altitudStr2 = altitud.ToString();     // Convertir a cadena de texto (string)

// Obtener el tipo de dato de las variables
Type tipoPosicion1 = posicionInt1.GetType();  // Tipo: System.Int32
Type tipoCalorias1 = caloriasFlt1.GetType();  // Tipo: System.Single (float)
Type tipoPeso1 = pesoStr1.GetType();          // Tipo: System.String
Type tipoAltitud1 = altitudStr1.GetType();    // Tipo: System.String

Type tipoPosicion2 = posicionInt2.GetType();  // Tipo: System.Int32
Type tipoCalorias2 = caloriasFlt2.GetType();  // Tipo: System.Single (float)
Type tipoPeso2 = pesoStr2.GetType();          // Tipo: System.String
Type tipoAltitud2 = altitudStr2.GetType();    // Tipo: System.String

Console.WriteLine(tipoPosicion1);
Console.WriteLine(tipoCalorias1);
Console.WriteLine(tipoPeso1);
Console.WriteLine(tipoAltitud1);

Console.WriteLine(tipoPosicion2);
Console.WriteLine(tipoCalorias2);
Console.WriteLine(tipoPeso2);
Console.WriteLine(tipoAltitud2);

Console.WriteLine("Mi nombre es " + nombre + ", tengo " + edad + " años y estoy a " + nivelDelMar + " metros sobre el nivel del mar, donde la temperatura es de " + temperatura + " grados Celsius. ¿Tengo coche? " + tieneCoche + ". Y por cierto, el resultado de una división que hice antes es de " + resultadoDivision + ".");

File: cursos/pria/src/python/1_variables.py
# Esto es un comentario de una línea

"""
   Esto es un comentario
   de múltiples líneas
"""

# VARIABLES
# Una variable es un símbolo que representa un valor que puede cambiar.
# En Python, simplemente asignamos un valor a una variable:
# nombre_variable = valor

edad = 25  # La edad es un número entero, no necesitamos especificar el tipo de dato


# TIPOS DE DATOS BÁSICOS

nivel_del_mar = 25                # Entero (int)
temperatura = -3.82               # Número real (float)
nombre = "Nacho López"            # Cadena de texto (str)
tiene_debian = True               # Booleano (bool): solo puede ser 'True' o 'False'


# OPERACIONES ARITMÉTICAS CON NÚMEROS

x = 5
y = 2

z = x + y          # Suma. Resultado: 7
z = x - y          # Resta. Resultado: 3
z = x * y          # Multiplicación. Resultado: 10
z = x / y          # División (siempre devuelve un 'float'). Resultado: 2.5
z = x // y         # División entera. Resultado: 2
z = x % y          # Módulo (resto de la división). Resultado: 1
z = z + 1          # Incrementa el valor de z en 1
z = z - 1          # Disminuye el valor de z en 1

# Puedes usar operadores abreviados (z += 1, z -= 1, z *= 2, z /= 3, etc):
z += 5             # Equivale a z = z + 5
z -= 2             # Equivale a z = z - 2


# OPERACIONES BÁSICAS CON CADENAS DE TEXTO (strings)

a = "GNU/"
b = "Linux"
c = a + b        # Concatenación. Resultado: "GNU/Linux"


# IMPRIMIR VARIABLES EN LA PANTALLA

print("¡Hola, mundo!")            # Imprime en la pantalla: ¡Hola, mundo!
print("El valor de x es", x)      # Imprime el valor de x

# Puedes imprimir en la pantalla cadenas de texto y variables
print("He comprado " + str(x) + " naranjas y " + str(y) + " limones.")
# O usando f-strings para mayor claridad (recomendado)
print(f"He comprado {x} naranjas y {y} limones.")


# CONVERSIÓN DE TIPOS DE DATOS

posicion = "5"                     # Una cadena que representa un número entero
calorias = "95.4"                  # Una cadena que representa un número real
peso = 85                          # Un número entero
altitud = -544.432                 # Un número real

# Convertir de cadena a entero
posicion_int1 = int(posicion)
print(f"Convertir 'posicion' de cadena a entero: {posicion_int1}")

# Convertir de cadena a número real (float)
calorias_flt1 = float(calorias)
print(f"Convertir 'calorias' de cadena a float: {calorias_flt1}")

# Convertir de número a cadena
peso_str1 = str(peso)
print(f"Convertir 'peso' de entero a cadena: '{peso_str1}'")

altitud_str1 = str(altitud)
print(f"Convertir 'altitud' de float a cadena: '{altitud_str1}'")

# Obtener el tipo de dato de las variables usando la función type()
tipo_posicion1 = type(posicion_int1)   # Tipo: <class 'int'>
tipo_calorias1 = type(calorias_flt1)   # Tipo: <class 'float'>
tipo_peso1 = type(peso_str1)           # Tipo: <class 'str'>
tipo_altitud1 = type(altitud_str1)     # Tipo: <class 'str'>

print(f"El tipo de 'posicion_int1' es {tipo_posicion1}")
print(f"El tipo de 'calorias_flt1' es {tipo_calorias1}")
print(f"El tipo de 'peso_str1' es {tipo_peso1}")
print(f"El tipo de 'altitud_str1' es {tipo_altitud1}")


# MENSAJE FINAL PARA EVITAR VARIABLES NO USADAS

print(f"Mi nombre es {nombre}, tengo {edad} años y estoy a {nivel_del_mar} metros sobre el nivel del mar, donde la temperatura es de {temperatura} grados Celsius. ¿Tengo Debian GNU/Linux instalado? {tiene_debian}. Y por cierto, el resultado de una división que hice antes es de {resultado_division}.")
